pills/17-nixpkgs-overriding-packages.xml

```diff
@@ -51,23 +51,20 @@
     </para>
     <screen><xi:include href="./17/fix-function.txt" parse="text" /></screen>
     <para>
-      It's a function that accepts a function <literal>f</literal>, calls <literal>f result</literal> on the result just returned by <literal>f result</literal> and returns it. In other words it's <literal>f(f(f(....</literal>
+      It's a function that accepts a function <literal>f</literal> and returns a function that recurses over f's arguments. Why is this useful? Because it plays nicely with Nix' laziness: it turns any function that otherwise would happily return (at least some) of its arguments unevaluated, into a function that recursively evaluates its arguments until there is nothing left to evaluate. To fix ideas (pun intended!) consider:
```

Comment:
> Why is this useful? Because it plays nicely with Nix' laziness

Okay, but that it is not the main reason, that’s just the reason it works even though it relies on infinite recursion.

Reply:
Yes, my phrasing is not optimal. Improving now.